R/geom-violin.r

```diff
@@ -105,11 +111,32 @@ GeomViolin <- ggproto("GeomViolin", Geom,
     newdata <- rbind(plyr::arrange(transform(data, x = xminv), y),
                      plyr::arrange(transform(data, x = xmaxv), -y))
 
+
     # Close the polygon: set first and last point the same
     # Needed for coord_polar and such
     newdata <- rbind(newdata, newdata[1,])
 
-    ggname("geom_violin", GeomPolygon$draw(newdata, ...))
+    # Draw quantiles if requested
+    if (length(draw_quantiles) > 0) {
+      dens <- cumsum(data$density) / sum(data$density)
+      ecdf <- approxfun(dens, data$y)
+      ys <- ecdf(draw_quantiles)
+
+      # Create list of path grobs for the quantile segments
+      quantile_grob_list <- lapply (ys, function(y_to_match) {
```

Comment:
Can you vectorise this so it returns a single grob (i.e. do the data manip inside `lapply()` and then call `GeomPath$draw()` only once? Otherwise it's going to be quite slow. 

I'd also recommend pulling out this code into a separate function. I think it should take `x` and `y` as inputs, and return a data frame of `x1`, `x2`, `y1`, `y2` suitable to be passed to `GeomSegment$draw()` (which I think is a better fit here).